Observer/SourceDeductionProcessor.php

```diff
@@ -87,6 +103,17 @@ public function execute(EventObserver $observer)
 
         $sourceSelectionResult = $this->getSourceSelectionResultFromOrder->execute($order);
 
+        if (!$extensionAttributes = $order->getExtensionAttributes()) {
+            $extensionAttributes = $this->orderExtensionFactory->create();
+        }
+
+        $extensionAttributes->setSources(
```

Comment:
Do we need to set the source results here? Are the extension attributes for this class/functionality used in the methods calling it? We just want to handle saving here realistically